lib/filterx/filterx-scope.c

```diff
@@ -436,3 +455,22 @@ filterx_scope_unref(FilterXScope *self)
   if (self && (g_atomic_counter_dec_and_test(&self->ref_cnt)))
     _free(self);
 }
+
+void
+filterx_scope_invalidate_log_msg_cache(FilterXScope *self)
+{
+  g_assert(filterx_scope_has_log_msg_changes(self));
+  gint i = 0;
+
+  while (i < self->variables->len)
+    {
+      FilterXVariable *v = &g_array_index(self->variables, FilterXVariable, i);
+
+      if (!filterx_variable_is_floating(v))
+        g_array_remove_index(self->variables, i);
```

Comment:
We had a short IRL discussion and agreed on going with this instead (because of performance).
As the variables won't be dropped as soon as soon they would be considered invalid, a neat trick is in the lookup logic to make sure it only returns still valid variables.